plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -61,6 +65,49 @@
 
     public static final String TYPE = "s3";
 
+    // prefix for aws client settings
+    private static final String PREFIX = "aws.config.";
```

Comment:
> I don't think the package names need to correspond in any way with the settings

Well. That has been our policy since the beginning. We even advertised that in our trainings IIRC. But I know it makes less sense now that logging is using the full package name. Still I always found super nice to know that if I want to debug a component which defines `x.y.z` setting, I can easily add DEBUG level to package named `x.y` before 5.0 and from 5.0 package `org.elasticsearch.x.y`.

@clintongormley Do you have any opinion on that? Consistent settings vs short setting names?

> I don't think it is necessary to have more than a single java package for a plugin.

I agree.